src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_header.tsx

```diff
@@ -6,36 +6,158 @@
  * your election, the "Elastic License 2.0", the "GNU Affero General Public
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
-import React from 'react';
-import { EuiButtonIcon, EuiFlexGroup, EuiSpacer, EuiTitle } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { distinctUntilChanged } from 'rxjs';
+
+import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { css } from '@emotion/react';
 import { i18n } from '@kbn/i18n';
 
-export const GridRowHeader = ({
-  isCollapsed,
-  toggleIsCollapsed,
-  rowTitle,
-}: {
-  isCollapsed: boolean;
-  toggleIsCollapsed: () => void;
-  rowTitle?: string;
-}) => {
-  return (
-    <div className="kbnGridRowHeader">
-      <EuiSpacer size="s" />
-      <EuiFlexGroup gutterSize="s">
-        <EuiButtonIcon
-          color="text"
-          aria-label={i18n.translate('kbnGridLayout.row.toggleCollapse', {
-            defaultMessage: 'Toggle collapse',
-          })}
-          iconType={isCollapsed ? 'arrowRight' : 'arrowDown'}
-          onClick={toggleIsCollapsed}
-        />
-        <EuiTitle size="xs">
-          <h2>{rowTitle}</h2>
-        </EuiTitle>
-      </EuiFlexGroup>
-      <EuiSpacer size="s" />
-    </div>
-  );
+import { GridLayoutStateManager } from '../types';
+import { deleteRow } from '../utils/row_management';
+import { DeleteGridRowModal } from './delete_grid_row_modal';
+import { GridRowTitle } from './grid_row_title';
+import { useGridRowHeaderStyles } from './use_grid_row_header_styles';
+
+export const GridRowHeader = React.memo(
```

Comment:
I think we should probably use `React.memo` in a few other places too, but I didn't want to muddy up this PR with changes unrelated to the main task. That being said, this one is necessary to prevent the arrow animation on open / close from being super jittery. Otherwise, whenever `GridRow` re-renders, its children components **also** all re-render (including `GridRowHeader`) which causes a noticeable lagginess in the animation.